### PR TITLE
[Feature][JDBC Sink] Add DM upsert support

### PR DESCRIPTION
**Pull request by lihjChina**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by EricJoy2048** [comment]
Please add e2e for this pr.

**Comment by lihjChina** [comment]
> Please add e2e for this pr.
complete~

**Comment by lihjChina** [comment]
done

**Review by Hisoka-X — DISMISSED** [review]
LGTM

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_dm_source_and_dm_upset_sink.conf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    url = "jdbc:dm://e2e_dmdb_upset:5236"
+    driver = "dm.jdbc.driver.DmDriver"
+    connection_check_timeout_sec = 1000
+    user = "SYSDBA"
+    password = "SYSDBA"
+    database = "SYSDBA"
+    primary_keys = [DM_BIT]
+    table = "E2E_TABLE_SINK_UPSET"
+    generate_sink_sql = true
+    support_upsert_by_query_primary_key_exist = true
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_dm_source_and_dm_upset_sink.conf]: remove `support_upsert_by_query_primary_key_exist `

Comment (davidzollo) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/resources/jdbc_dm_source_and_dm_upset_sink.conf]: solved

**Review by davidzollo — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**